### PR TITLE
test material text field example

### PR DESCRIPTION
**Pull request by derdilla**
Adds simple tests to the last two [TextField](https://api.flutter.dev/flutter/material/TextField-class.html) examples as part of fixing #130459. 
 
## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [X] All existing and new tests are passing.

### CODE REVIEW

File: examples/api/test/material/text_field/text_field.1_test.dart [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/material/text_field/text_field.1.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('Dialoge shows submitted TextField value', (WidgetTester tester) async {
```
Comment (bleroux) [inline examples/api/test/material/text_field/text_field.1_test.dart:10]:
```suggestion
  testWidgets('Dialog shows submitted TextField value', (WidgetTester tester) async {
```

File: examples/api/test/material/text_field/text_field.2_test.dart [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    await tester.pumpWidget(const example.TextFieldExamplesApp());
+
+    expect(find.text('TextField Examples'), findsOneWidget);
+    expect(find.byType(TextField), findsNWidgets(2));
+    expect(find.byType(example.FilledTextFieldExample), findsOneWidget);
+    expect(find.byType(example.OutlinedTextFieldExample), findsOneWidget);
+
+    final TextField filled = tester.widget<TextField>(find.descendant(
+        of: find.byType(example.FilledTextFieldExample),
+        matching: find.byType(TextField)));
```
Comment (bleroux) [inline examples/api/test/material/text_field/text_field.2_test.dart:20]:
```suggestion
    final TextField filled = tester.widget<TextField>(find.descendant(
      of: find.byType(example.FilledTextFieldExample),
      matching: find.byType(TextField),
    ));
```

**Review by bleroux — APPROVED** [review]
LGTM with small nits.
Thanks for your contribution!

File: examples/api/test/material/text_field/text_field.2_test.dart [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    await tester.pumpWidget(const example.TextFieldExamplesApp());
+
+    expect(find.text('TextField Examples'), findsOneWidget);
+    expect(find.byType(TextField), findsNWidgets(2));
+    expect(find.byType(example.FilledTextFieldExample), findsOneWidget);
+    expect(find.byType(example.OutlinedTextFieldExample), findsOneWidget);
+
+    final TextField filled = tester.widget<TextField>(find.descendant(
+      of: find.byType(example.FilledTextFieldExample),
+      matching: find.byType(TextField)));
```
Comment (bleroux) [inline examples/api/test/material/text_field/text_field.2_test.dart:20]: Missing comma at the end of `matching`parameter line and the two closing parentheses should be on a new line, see:
```suggestion
    final TextField filled = tester.widget<TextField>(find.descendant(
      of: find.byType(example.FilledTextFieldExample),
      matching: find.byType(TextField),
    ));
```

File: examples/api/test/material/text_field/text_field.2_test.dart [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    expect(filled.decoration!.suffixIcon, isA<Icon>()
+        .having((Icon icon) => icon.icon, 'icon', Icons.clear));
+    expect(filled.decoration!.labelText, 'Filled');
+    expect(filled.decoration!.hintText, 'hint text');
+    expect(filled.decoration!.helperText, 'supporting text');
+    expect(filled.decoration!.filled, true);
+
+    final TextField outlined = tester.widget<TextField>(find.descendant(
+      of: find.byType(example.OutlinedTextFieldExample),
+      matching: find.byType(TextField)));
```
Comment (bleroux) [inline examples/api/test/material/text_field/text_field.2_test.dart:32]: Same

**Review by bleroux — CHANGES_REQUESTED** [review]
Two small formatting issues.

File: examples/api/test/material/text_field/text_field.2_test.dart (hunk 3 above)
Comment (NobodyForNothing) [reply examples/api/test/material/text_field/text_field.2_test.dart]: thanks

**Review by bleroux — APPROVED** [review]
LGTM! Thanks again for this great addition. 🙏

**Review by Renzo-Olivares — APPROVED** [review]

**Pull request merged**